app/components/Sidebar/components/CollectionLink.js

```diff
@@ -34,61 +31,63 @@ class CollectionLink extends React.Component<Props> {
       activeDocument,
       prefetchDocument,
       canUpdate,
       ui,
     } = this.props;
+
     const expanded = collection.id === ui.activeCollectionId;
 
     return (
-      <DropToImport
-        key={collection.id}
-        collectionId={collection.id}
-        activeClassName="activeDropZone"
-      >
-        <SidebarLink
+      <>
+        <DropToImport
           key={collection.id}
-          to={collection.url}
-          icon={<CollectionIcon collection={collection} expanded={expanded} />}
-          iconColor={collection.color}
-          expanded={expanded}
-          hideDisclosure
-          menuOpen={this.menuOpen}
-          label={
-            <EditableTitle
-              title={collection.name}
-              onSubmit={this.handleTitleChange}
-              canUpdate={canUpdate}
-            />
-          }
-          exact={false}
-          menu={
-            <CollectionMenu
-              position="right"
-              collection={collection}
-              onOpen={() => (this.menuOpen = true)}
-              onClose={() => (this.menuOpen = false)}
-            />
-          }
+          collectionId={collection.id}
+          activeClassName="activeDropZone"
         >
-          <Flex column>
-            {collection.documents.map((node) => (
-              <DocumentLink
-                key={node.id}
-                node={node}
-                documents={documents}
-                collection={collection}
-                activeDocument={activeDocument}
-                prefetchDocument={prefetchDocument}
+          <SidebarLink
+            key={collection.id}
+            to={collection.url}
+            icon={
+              <CollectionIcon collection={collection} expanded={expanded} />
+            }
+            iconColor={collection.color}
+            expanded={expanded}
+            hideDisclosure
+            menuOpen={this.menuOpen}
+            label={
+              <EditableTitle
+                title={collection.name}
+                onSubmit={this.handleTitleChange}
                 canUpdate={canUpdate}
-                depth={1.5}
               />
-            ))}
-          </Flex>
-        </SidebarLink>
-      </DropToImport>
+            }
+            exact={false}
+            menu={
+              <CollectionMenu
+                position="right"
+                collection={collection}
+                onOpen={() => (this.menuOpen = true)}
+                onClose={() => (this.menuOpen = false)}
+              />
+            }
+          ></SidebarLink>
+        </DropToImport>
+
+        {expanded &&
+          collection.documents.map((node) => (
+            <DocumentLink
+              key={node.id}
+              node={node}
+              collection={collection}
+              activeDocument={activeDocument}
+              prefetchDocument={prefetchDocument}
+              canUpdate={canUpdate}
+              depth={1.5}
+            />
+          ))}
+      </>
```

Comment:
🍖 nested/recursive calls to Collection and Document Link do not have nested HTML elements